components/ai_chat/core/browser/ai_chat_metrics.cc

```diff
@@ -349,6 +513,83 @@ void AIChatMetrics::ReportChatCounts() {
   // TODO(djandries): remove the following report when Nebula experiment is over
   p3a_utils::RecordToHistogramBucket(kChatCountNebulaHistogramName,
                                      kChatCountBuckets, chat_count);
+
+  uint64_t max_chat_duration =
+      chat_durations_storage_.GetHighestValueInPeriod();
+  p3a_utils::RecordToHistogramBucket(kMaxChatDurationHistogramName,
+                                     kMaxChatDurationBuckets,
+                                     max_chat_duration);
+
+  uint64_t chat_with_history_count =
+      chat_with_history_count_storage_.GetWeeklySum();
+  int history_percentage = static_cast<int>(std::ceil(
+      chat_count > 0
+          ? (static_cast<double>(chat_with_history_count) / chat_count) * 100.0
+          : 0.0));
+
+  p3a_utils::RecordToHistogramBucket(kChatHistoryUsageHistogramName,
+                                     kChatHistoryUsageBuckets,
+                                     history_percentage);
+}
+
+void AIChatMetrics::RecordContextSource(ConversationHandlerForMetrics* handler,
+                                        mojom::ConversationTurnPtr& entry) {
+  ContextSource context = ContextSource::kTextInputWithoutPage;
+  if (prompted_via_omnibox_) {
+    context = ContextSource::kOmniboxInput;
+  } else if (entry->text ==
+                 l10n_util::GetStringUTF8(IDS_CHAT_UI_SUMMARIZE_PAGE) ||
```

Comment:
I think instead of comparing the `text`, we can more reliably compare the action:
```
entry->action_type == mojom::ActionType::SUMMARIZE_PAGE
```

Reply:
i tried this at first, but noticed that the action type would sometimes be `UNSPECIFIED`

Reply:
ah yes that's because "summarize this page" is a suggestion. I think the `mojom::ActionType::SUMMARIZE_PAGE` is only used for pre-opted-in users. Let me try to find a fix...